### DISCUSSION TITLE
32bit systems

### DISCUSSION THREAD

**Discussion by MacOS-9** [android/ndk#1667] [Q&A]
is there any new/old versions that support 32bit systems, and if not can i compile it for 32bit myself?

**Comment by DanAlbert** [comment]
New versions no. Old versions, very old and you shouldn't use them in production apps. You can _try_ to build for 32-bit, but you have to build the NDK and all its dependencies. https://android.googlesource.com/platform/ndk/+/master/docs/Onboarding.md explains how to get started but it's going to be very difficult. You will need to bootstrap everything because the compilers we use to build the NDK and its dependencies aren't 32-bit either.

Are you really using macOS 9? We also don't support PowerPC so that's going to be yet another layer of trouble.

**Comment by enh-google** [comment]
> is there any new/old versions that support 32bit systems, and if not can i compile it for 32bit myself?

just to be clear: you're asking about 32-bit *hosts*, right? because the latest NDK supports 32-bit *devices* and there are no current plans to remove that support. but i'm not sure we *ever* supported 32-bit mac or linux... they were already obsolete before we started!
